servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/NettyHttpServerTest.java

```diff
@@ -601,14 +595,22 @@ void testErrorBeforeRead(ExecutorSupplier clientExecutorSupplier,
 
             final BlockingIterator<Buffer> httpPayloadChunks = response.payloadBody().toIterable().iterator();
 
-            Exception e = assertThrows(Exception.class, () -> httpPayloadChunks.next());
+            Exception e = assertThrows(Exception.class, httpPayloadChunks::next);
             assertThat(e, either(instanceOf(RuntimeException.class)).or(instanceOf(ExecutionException.class)));
             // Due to a race condition, the exception cause here can vary.
             // If the socket closure is delayed slightly
             // (for example, by delaying the Publisher.error(...) on the server)
             // then the client throws ClosedChannelException. However if the socket closure happens quickly enough,
             // the client throws NativeIoException (KQueue) or IOException (NIO).
             assertThat(e.getCause(), instanceOf(IOException.class));
+        } catch (Throwable cause) {
+            // The server intentionally triggers an error when it writes, if this happens before all content is read
+            // the client may fail to write the request due to premature connection closure.
+            if (cause instanceof ExecutionException) {
```

Comment:
Alternative approach can be to use "flushOnEnd" strategy by converting `StreamingHttpRequest` to an aggregated one and back. Then we don't need to worry about write failures on the client-side.
This approach also good. Either way is great to fix this test.